multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -414,7 +418,10 @@ private HttpResponse<InputStream> callRemoteEndpointWithRetry(HttpClient client,
             var request = buildFetchFileRequest(decodedUrl);
             LOGGER.debug(Messages.CALLING_REMOTE_MTAR_ENDPOINT, request.uri());
             var response = client.send(request, BodyHandlers.ofInputStream());
-            if (response.statusCode() / 100 != 2) {
+            if (response.statusCode() == 401) {
+                String errorMessage = MessageFormat.format(Messages.DEPLOY_FROM_URL_WRONG_CREDENTIALS, UriUtil.stripUserInfo(decodedUrl));
+                throw new SLException(errorMessage);
```

Comment:
just in case, extract: String error = readErrorBodyFromResponse(response);
and log it